pandas/core/generic.py

```diff
@@ -10600,6 +10600,13 @@ def mad(self, axis=None, skipna=None, level=None):
         {examples}
         """
         if skipna is None:
+            warnings.warn(
+                "Passing None for skipna is deprecated and will raise in a future"
+                "version. Pass True instead. Only boolean values will be allowed "
```

Comment:
Maybe good to add a separate `if not is_bool` to warn on non boolean values so it's easier to enforce in the future.

Reply:
Done

Reply:
Nit: Could you use `pandas.core.dtypes.common.is_bool`

Reply:
Done